libraries/bot-azure/src/main/java/com/microsoft/bot/azure/queues/AzureQueueStorage.java

```diff
@@ -0,0 +1,90 @@
+import com.microsoft.bot.schema.Activity;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Base64;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Service used to add messages to an Azure.Storage.Queues.
+ */
+public class AzureQueueStorage extends QueueStorage {
+    private Boolean createQueueIfNotExists = true;
+    private final QueueClient queueClient;
+
+    /**
+     * Initializes a new instance of the {@link AzureQueueStorage} class.
+     * @param queuesStorageConnectionString Azure Storage connection string.
+     * @param queueName Name of the storage queue where entities will be queued.
+     */
+    public AzureQueueStorage(String queuesStorageConnectionString, String queueName) {
+        if (StringUtils.isBlank(queuesStorageConnectionString)) {
+            throw new IllegalArgumentException("queuesStorageConnectionString is required.");
+        }
+
+        if (StringUtils.isBlank(queueName)) {
+            throw new IllegalArgumentException("queueName is required.");
+        }
+
+        queueClient = new QueueClientBuilder()
+            .connectionString(queuesStorageConnectionString)
+            .queueName(queueName)
+            .buildClient();
+    }
+
+    /**
+     * Queue an Activity to an Azure.Storage.Queues.QueueClient.
+     * The visibility timeout specifies how long the message should be invisible
+     * to Dequeue and Peek operations. The message content must be a UTF-8 encoded string that is up to 64KB in size.
+     * @param activity This is expected to be an {@link Activity} retrieved from a call to
+     *                 activity.GetConversationReference().GetContinuationActivity().
+     *                 This enables restarting the conversation using BotAdapter.ContinueConversationAsync.
+     * @param visibilityTimeout Default value of 0. Cannot be larger than 7 days.
+     * @param timeToLive Specifies the time-to-live interval for the message.
+     * @return {@link SendMessageResult} as a Json string, from the QueueClient SendMessageAsync operation.
+     */
+    @Override
+    public CompletableFuture<String> queueActivity(Activity activity,
+                                                   @Nullable Duration visibilityTimeout,
+                                                   @Nullable Duration timeToLive) {
+        return CompletableFuture.supplyAsync(() -> {
+            if (createQueueIfNotExists) {
+                // This is an optimization flag to check if the container creation call has been made.
+                // It is okay if this is called more than once.
+                createQueueIfNotExists = false;
+                try {
+                    queueClient.create();
+                } catch (QueueStorageException e) {
+                    e.printStackTrace();
```

Comment:
Is it acceptable for the code to continue at this point or should we return here?

Reply:
Hi @LeeParrishMSFT, we improved this behaviour to throw a `RuntimeException` like the [CosmosDbPartitionedStorage](https://github.com/microsoft/botbuilder-java/blob/main/libraries/bot-azure/src/main/java/com/microsoft/bot/azure/CosmosDbPartitionedStorage.java#L285-L289) does.